File: urionlinejudge/1061.go
//
// For some reason this code in Golang got TLE, but the C++ solution has accepted
//

package main

import "fmt"

const DayInSeconds = 86400
const HourInSeconds = 3600
const MinuteInSeconds = 60

func getSecondsFromDays(days int) int {
	return days * DayInSeconds
}

func getSecondsFromHours(hours int) int {
	return hours * HourInSeconds;
}

func getSecondsFromMinutes(minutes int) int {
	return minutes * MinuteInSeconds
}

func main() {
	var initialDay, finalDay, initialHour, finalHour, initialMinutes, finalMinutes, initialSeconds, finalSeconds int;

	var dayLabel string
	fmt.Scanf("%s %d", &dayLabel, &initialDay);
	fmt.Scanf("%d : %d : %d",&initialHour, &initialMinutes, &initialSeconds)

	totalInitialSeconds := getSecondsFromDays(initialDay) + getSecondsFromHours(initialHour) + getSecondsFromMinutes(initialMinutes) + initialSeconds	

	fmt.Scanf("%s %d", &dayLabel, &finalDay);
	fmt.Scanf("%d : %d : %d", &finalHour, &finalMinutes, &finalSeconds)	
	totalFinalSeconds := getSecondsFromDays(finalDay) + getSecondsFromHours(finalHour) + getSecondsFromMinutes(finalMinutes) + finalSeconds	

	diffSeconds := totalFinalSeconds - totalInitialSeconds;

	days := diffSeconds / DayInSeconds
	fmt.Println(days, "dia(s)")
	
	rest := diffSeconds % DayInSeconds
	hours := rest / HourInSeconds
	fmt.Println(hours, "hora(s)")

	rest = diffSeconds % HourInSeconds
	minutes := rest / MinuteInSeconds
	fmt.Println(minutes, "minuto(s)")

	rest = diffSeconds % MinuteInSeconds
	fmt.Println(rest, "segundo(s)")

}

File: urionlinejudge/beginner/1061.cpp
#include <stdio.h>

const int DayInSeconds = 86400;
const int HourInSeconds = 3600;
const int MinuteInSeconds = 60;

int getSecondsFromDays(int days) {
	return days * DayInSeconds;
}

int getSecondsFromHours(int hours) {
	return hours * HourInSeconds;
}

int getSecondsFromMinutes(int minutes) {
	return minutes * MinuteInSeconds;
}

int main() {
	int initialDay, finalDay, initialHour, finalHour, initialMinutes, finalMinutes, initialSeconds, finalSeconds;

    char dayLabel[3];
	scanf("%s %d", dayLabel, &initialDay);
	scanf("%d : %d : %d", &initialHour, &initialMinutes, &initialSeconds);

	int totalInitialSeconds = getSecondsFromDays(initialDay) + getSecondsFromHours(initialHour) + getSecondsFromMinutes(initialMinutes) + initialSeconds;

	scanf("%s %d", dayLabel, &finalDay);
	scanf("%d : %d : %d", &finalHour, &finalMinutes, &finalSeconds);
	int totalFinalSeconds = getSecondsFromDays(finalDay) + getSecondsFromHours(finalHour) + getSecondsFromMinutes(finalMinutes) + finalSeconds;	

	int diffSeconds = totalFinalSeconds - totalInitialSeconds;

	int days = diffSeconds / DayInSeconds;
	printf("%d dia(s)\n", days);
	
	int rest = diffSeconds % DayInSeconds;
	int hours = rest / HourInSeconds;
	printf("%d hora(s)\n", hours);

	rest = diffSeconds % HourInSeconds;
	int minutes = rest / MinuteInSeconds;
	printf("%d minuto(s)\n", minutes);

	rest = diffSeconds % MinuteInSeconds;
	printf("%d segundo(s)\n", rest);

    return 0;
}
